src/main/java/org/folio/linked/data/validation/entity/PrimaryTitleEntityValidator.java

```diff
@@ -0,0 +1,36 @@
+package org.folio.linked.data.validation.entity;
+
+import static java.util.Objects.nonNull;
+import static java.util.Optional.ofNullable;
+import static org.folio.ld.dictionary.PropertyDictionary.MAIN_TITLE;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.INSTANCE;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.TITLE;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.WORK;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.folio.linked.data.model.entity.Resource;
+import org.folio.linked.data.model.entity.ResourceEdge;
+import org.folio.linked.data.validation.PrimaryTitleConstraint;
+
+public class PrimaryTitleEntityValidator implements ConstraintValidator<PrimaryTitleConstraint, Resource> {
+
+  @Override
+  public boolean isValid(Resource resource, ConstraintValidatorContext context) {
+    if (!resource.isOfType(INSTANCE, WORK)) {
+      return true;
+    }
+    return ofNullable(resource.getOutgoingEdges())
```

Comment:
suggest to simplify
```
if (Collections.isEmty(resource.getOutgoingEdges()) {
   return false
}
resource.getOutgoingEdges()
    stream()
...
```

Reply:
Optional class was introduced to let us avoid such constructions, and I like it more. Anyway there is minimal difference, so let me leave it as is or provide some proof please :)

Reply:
the problem in the complicated "map" function, if you want to save your way, It will be better to move it to a separate method
that's why I suggest it to simplify by extra check

Reply:
alright, changed